src/c/functions.c

```diff
@@ -154,57 +158,55 @@ void compute_forces(const double dt, const size_t particles_size, const size_t c
-    velocities[i].y_component = velocities[i].y_component + accelerations[i].y_component * dt;
-  }
+inline void compute_velocity(const double dt, const size_t particle_index,
+                             const Vector *accelerations, Vector *velocities) {
+  velocities[particle_index].x_component =
+    velocities[particle_index].x_component + accelerations[particle_index].x_component * dt;
+  velocities[particle_index].y_component =
+    velocities[particle_index].y_component + accelerations[particle_index].y_component * dt;
 }
 
 /**
  * Computes the displacement of the particles,
  * with an applied velocity for a given time delta.
  */
-void compute_displacement(const double dt, const size_t size, const Vector *velocities, Vector *displacements) {
-  for (size_t i = 0; i < size; ++i) {
-    displacements[i].x_component = displacements[i].x_component + velocities[i].x_component * dt;
-    displacements[i].y_component = displacements[i].y_component + velocities[i].y_component * dt;
-  }
+inline void compute_displacement(const double dt, const size_t particle_index,
+                                 const Vector *velocities, Vector *displacements) {
+  displacements[particle_index].x_component =
+    displacements[particle_index].x_component + velocities[particle_index].x_component * dt;
+  displacements[particle_index].y_component =
+    displacements[particle_index].y_component + velocities[particle_index].y_component * dt;
 }
 
 /**
  * Displace all particles given their displacements.
  */
-void displace_particles(const size_t size, const Vector *displacements, Particle *particles) {
-  for (size_t i = 0; i < size; ++i) {
-    Particle *particle = &particles[i];
-    particle->x_coordinate += (displacements[i].x_component * 1000);
-    particle->y_coordinate += (displacements[i].y_component * 1000);
-  }
+inline void displace_particle(const size_t particle_index, const Vector *displacements,
+                              Particle *particles) {
+  Particle *particle = &particles[particle_index];
+  particle->x_coordinate += (displacements[particle_index].x_component * 1000);
+  particle->y_coordinate += (displacements[particle_index].y_component * 1000);
 }
 
 /**
   * Changes the displacement if the new position would surpass the Y limit.
   */
-void fix_displacements(const size_t size, Vector *velocities, Particle *particles){
-  for(size_t i = 0; i < size; ++i){
-    Particle *particle = &particles[i];
-
-    double diff = particle->y_coordinate - particle->radius;
-    if(diff < 0){
-      particle->y_coordinate = particle->radius;
-      velocities[i].y_component = 0;
-    }
+inline void fix_displacement(const size_t particle_index, Vector *velocities, Particle *particles) {
+  Particle *particle = &particles[particle_index];
+
+  double diff = particle->y_coordinate - particle->radius;
+  if (diff < 0) {
+    particle->y_coordinate = particle->radius;
```

Comment:
I think the assignment is very pointless since it is not reused, I would just:

```c
particles[particle_index].y_coordinate = particles[particle_index].radius;
```

_(Also applies to `displace_particle` above)_.
```

Reply:
Fixed